airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -94,8 +97,8 @@ def __init__(
         self.share_name = share_name
         self.directory_path = directory_path
         self.directory_name = directory_name
-        if self.directory_path is None:
-            self.directory_path = directory_name
+        if self.directory_path is None and self.directory_name is not None:
```

Comment:
Technically `directory_path` (and `directory_name`) are optional. The warning would be emitted even if a user purposely did not wish to specify `directory_name`.